pkgs/development/python-modules/lightgbm/default.nix

```diff
@@ -28,16 +34,25 @@ buildPythonPackage rec {
 
   dontUseCmakeConfigure = true;
 
-  buildInputs = lib.optionals stdenv.cc.isClang [
+  buildInputs = (optionals stdenv.cc.isClang [
     llvmPackages.openmp
-  ];
+  ]) ++ (optionals gpuSupport [
+    boost
+    ocl-icd
+    opencl-headers
+  ]);
```

Comment:
```
  buildInputs = lib.optionals stdenv.cc.isClang [
    llvmPackages.openmp
  ] ++ lib.optionals gpuSupport [
    boost
    ocl-icd
    opencl-headers
  ];
```

Reply:
Is there any reason for adding lib. in multiple places rather than using `with lib;`?

Reply:
with lib should not be used over the entire file, as this is an anti pattern. Also using it locally here does not generate much cleaner code.

Reply:
Feedback of the form "X is an anti pattern" is an anti pattern. Please mention any specific concerns you have. I understand that multiple overlapping `with` statements can cause confusion, but the way it is done in this and many other packages is very clear.

I did miss the `meta = with lib;` below. I can drop that and update the PR. I can instead drop the top level `with lib;` if you insist, but I would like to know why.

Reply:
Removed `meta = with lib;`. Marking as resolved. Please reopen if you have anything more to add.